relate/src/main/scala/com/lucidchart/relate/SqlResult.scala

```diff
@@ -50,12 +50,8 @@ class SqlResult(val resultSet: java.sql.ResultSet) extends CollectionsSqlResult
   }
 
   def asScalar[A](): A = asScalarOption.get
-  def asScalarOption[A](): Option[A] = {
-    if (resultSet.next()) {
-      Some(resultSet.getObject(1).asInstanceOf[A])
-    } else {
-      None
-    }
+  def asScalarOption[A](): Option[A] = withResultSet { resultSet =>
```

Comment:
Can we make this `asCollection[A, Option](_.asInstanceOf[A])`?

Reply:
Honestly, I'd rather go the opposite direction, and make `asSingle` and `asSingleOption` _stop_ using `asCollection`